### ISSUE TITLE
Hyperparameters for 48K sample rate dataset. 

### ISSUE THREAD

**Issue by SRHyd** [Rayhane-mamah/Tacotron-2#70]
Hello,

Has anyone tried this repo with 48K sample rate dataset?

I am training English(Indian accent) dataset which has 48000-16bit-Mono sample rate wave files. I have modified the following hyperparameters for the dataset in hparams.py
 
------------------------------------------------------------------------------------------
modified hyperparameters in hparams.py
------------------------------------------------------------------------------------------

use_all_gpus = True,  #False
num_gpus = 8,  #1

num_freq = 1201, #513 ()
use_lws=False, #True

#Mel spectrogram
n_fft = 2400, #Extra window size is filled with 0 paddings to match this parameter
hop_size = 600, # For 48000, 600~=12.5ms, For 22050Hz, 275 ~= 12.5 ms
win_size = 2400, # For 48000, 2400 ~= 50ms, For 22050Hz, 1100 ~= 50 ms (If None, win_size = n_fft)
sample_rate = 48000, #48000(my dataset), 22050 Hz (corresponding to ljspeech dataset)
frame_shift_ms = None,

outputs_per_step = 1,

prenet_layers = [128, 128], #number of layers and number of units of prenet
decoder_layers = 2, #number of decoder lstm layers
decoder_lstm_units = 512, #number of decoder lstm units on each layer
max_iters = 2500, #Max decoder steps during inference (Just for safety from infinite loop cases)

mask_encoder = False, #whether to mask encoder padding while computing attention
mask_decoder = False, #Whether to use loss mask for padded sequences 

predict_linear = True, #Whether to add a post-processing network to the Tacotron to predict linear spectrograms (True mode Not tested!!)

tacotron_batch_size = 16, #number of training samples on each training steps

-------------------------------------------------------

After 25K steps of training, the loss is at 0.75474, and the alignment plot is like this
 
![step-25000-align](https://user-images.githubusercontent.com/40122465/41400207-1f813ba4-6fda-11e8-8946-a1ca21a93c72.png)

------------------------
The mel-spectrogram is like this

![step-25000-mel-spectrogram](https://user-images.githubusercontent.com/40122465/41399730-b89e3a28-6fd8-11e8-92f1-7bc5d6240df5.png)

-------------------------------------------------------

To me the mel-spectrogram looks fine. The audio samples are fine but robo voice. The alignment plot seems different than others seen in this repo? What's wrong in alignment plot?

Regards,
SRHyd

**Comment by begeekmyfriend** [comment]
I am not sure 48KHz is OK for Tacotron or not. I suggest you can transfer all you wav files into 22050KHz by `FFmpeg` and try again. For me 16KHz is all right.
Your hyper parameter seems no question except that the `n_fft` is not power of 2. I am not sure this would be a problem. So please try resampling into 22050KHz. If it still fails to convergence, then there might be problems in your transcript symbols.

**Comment by SRHyd** [comment]
@begeekmyfriend,  Thanks for reaching out. I will try with the sample_rate=16000, and param as below

nfft = 1024
hop_size = 256,
win_size = 1024,
sample_rate = 16000,
frame_shift_ms = None,

# trim params
trim_fft_size = 512,
trim_hop_size = 128,
outputs_per_step = 1,
 
Also,  I want to use LWS for faster inversion of the spectrogram. I will not be using the wavenet vocoder.  
use_lws=True # Does not work if n_ffit is not multiple of hop_size!!'
Will it work?

Why is it required to transfer all wave file to 16000KHz? I think the resampling is done by default in audio.py during the preprocessing. (Please refer to audio.py in line 10,  librosa.core.load(path, sr=sr).

**Comment by azraelkuan** [comment]
try set outputs_per_step =3, where encoder steps is close to decode step

**Comment by SRHyd** [comment]
Thank you @azraelkuan. 
I have changed outputs_per_step=3, now I am getting the alignment after 6K steps.

![step-6000-align](https://user-images.githubusercontent.com/40122465/41496796-e5639bdc-7165-11e8-883b-3b3bd67e8c36.png)

**Comment by Rayhane-mamah** [comment]
Hello, bigger "outputs_per_step" indeed makes alignments easier, but your batch size was small that was the problem. Also if you want to use 48kHz, you should the following parameters.


	num_freq = 2049, # (= n_fft / 2 + 1) only used when adding linear spectrograms post processing network
#Mel spectrogram
	n_fft = 4096, #Extra window size is filled with 0 paddings to match this parameter
	hop_size = 600, #For 22050Hz, 275 ~= 12.5 ms
	win_size = 2400, #For 22050Hz, 1100 ~= 50 ms (If None, win_size = n_fft)
	sample_rate = 48000, #22050 Hz (corresponding to ljspeech dataset)
	frame_shift_ms = None,

This keeps consistency with the 12.5ms 50ms window and shift. You can also set n_fft = 2400 and num_freq=1201. There's no problem with that as well.

Closing the issue, feel free to reopen if there is anything else :)
